README.rst

```diff
@@ -164,6 +164,92 @@ a tuple of (``status``, ``headers``, ``body``).
             '728d329e-0e86-11e4-a748-0c84dc037c13'
         )
 
+You can also pass a compiled regex to `add_callback` to match multiple urls:
+
+..  code-block:: python
+
+    import re, json
+
+    from functools import reduce
+
+    import responses
+    import requests
+
+    operators = {
+      'sum': lambda x, y: x+y,
+      'prod': lambda x, y: x*y,
+      'pow': lambda x, y: x**y
+    }
+
+    @responses.activate
+    def test_regex_url():
+
+        def request_callback(request):
+            payload = json.loads(request.body)
+            operator_name = request.path_url[1:]
+
+            operator = operators[operator_name]
+            if not operator:
+              raise Exception('Unexpected operator: ' + operator)
+
+            resp_body = {'value': reduce(operator, payload['numbers'])}
+            headers = {'request-id': '728d329e-0e86-11e4-a748-0c84dc037c13'}
+            return (200, headers, json.dumps(resp_body))
+
+        responses.add_callback(
+            responses.POST,
+            re.compile('http://calc.com/[sum|prod|pow|unsupported]'),
+            callback=request_callback,
+            content_type='application/json',
+        )
+
+        resp = requests.post(
+            'http://calc.com/sum',
+            json.dumps({'numbers': [2, 3, 4]}),
+            headers={'content-type': 'application/json'},
+        )
+        assert resp.json() == {'value': 9}
+        print('Test of sum passed')
+
+        resp = requests.post(
+            'http://calc.com/prod',
+            json.dumps({'numbers': [2, 3, 4]}),
+            headers={'content-type': 'application/json'},
+        )
+        assert resp.json() == {'value': 24}
+        print('Test of prod passed')
+
+        resp = requests.post(
+            'http://calc.com/pow',
+            json.dumps({'numbers': [2, 3, 4]}),
+            headers={'content-type': 'application/json'},
+        )
+        assert resp.json() == {'value':4096}
+        print('Test of pow passed')
+
+        try:
+          resp = requests.post(
+              'http://calc.com/unsupported',
+              json.dumps({'numbers': [2, 3, 4]}),
+              headers={'content-type': 'application/json'},
+          )
+          assert False, "Expecting an exception here"
+        except Exception:
+          print('Test of unsupported passed')
+
+        try:
+          resp = requests.post(
+                'http://calc.com/foobar',
+                json.dumps({'numbers': [2, 3, 4]}),
+                headers={'content-type': 'application/json'},
+          )
+          assert False, "Expecting an exception here"
+        except requests.exceptions.ConnectionError:
+          print('Test of unhandled path passed')
```

Comment:
Could we include only one sample usage instead of all the permutations? This snippet is a bit long right now.

Reply:
Sure thing! Will amend.